Analysis/Tasks/PWGHF/taskD0D0barCorrelation.cxx

```diff
@@ -0,0 +1,875 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskD0D0barCorrelation.cxx
+/// \brief D0-D0bar analysis task - data-like, MC-reco and MC-kine analyses. For ULS and LS pairs
+///
+/// \author Fabio Colamaria <fabio.colamaria@ba.infn.it>, INFN Bari
+
+#include <cmath>
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+using namespace o2::framework::expressions;
+using namespace o2::constants::math;
+
+// the following is needed to extend the standard candidate table, and allow grouping candidate table by collisions
+namespace o2::aod
+{
+namespace hf_2prong_correlation
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+} // namespace hf_2prong_correlation
+DECLARE_SOA_TABLE(HF2ProngCollis, "AOD", "COLLID_2PR", aod::hf_2prong_correlation::CollisionId);
+
+using Big2Prong = soa::Join<aod::HfCandProng2, aod::HFSelD0Candidate, aod::HF2ProngCollis>;
+using Big2ProngMC = soa::Join<aod::HfCandProng2, aod::HFSelD0Candidate, aod::HfCandProng2MCRec, aod::HF2ProngCollis>;
+} // namespace o2::aod
+
+// preliminary task to fill the column index to the extended candidate table
+struct CreateBig2Prong {
+
+  Produces<aod::HF2ProngCollis> create2ProngIndexCollColumn;
+  void process(aod::HfCandProng2 const& candidates, aod::Tracks const& tracks)
+  {
+    for (auto& candidate : candidates) {
+      int indexColl = candidate.index0_as<aod::Tracks>().collisionId(); //takes index of collision from first D daughter
+      create2ProngIndexCollColumn(indexColl);
+    }
+  }
+};
```

Comment:
This should be removed once https://github.com/AliceO2Group/AliceO2/pull/5469 gets merged.

Reply:
Sure, for now I left it in place

Reply:
Since #5469 was merged this morning, I adapted the task and committed the updated version